### PR TITLE
Rebased DI patch

### PR DESCRIPTION
**Pull request by Aaronontheweb**
Patch to Issues with DI Support #706

Rebased @jcwrequests's work on dev and cleaned up the DI samples. Patch is in a state where it can be reviewed.

Might break this PR after I merge in @HCanber's and I's big PRs, but no worries - I'll update it.

cc @akkadotnet/developers I need some help reviewing this before V1.0 ships pretty please

### CODE REVIEW

File: src/contrib/dependencyInjection/Examples/BasicAutoFacUses/Actors.cs~HEAD [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿﻿using Akka.Actor;
```
Comment (rogeralsing) [inline src/contrib/dependencyInjection/Examples/BasicAutoFacUses/Actors.cs~HEAD]: src/contrib/dependencyInjection/Examples/BasicAutoFacUses/Actors.cs~HEAD

Git conflict file. remove

File: src/contrib/dependencyInjection/Examples/BasicAutoFacUses/BasicAutoFacUses.csproj~HEAD [hunk 2]
```diff
@@ -0,0 +1,108 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (rogeralsing) [inline src/contrib/dependencyInjection/Examples/BasicAutoFacUses/BasicAutoFacUses.csproj~HEAD]: another conflict file

File: src/contrib/dependencyInjection/Examples/BasicAutoFacUses/Program.cs~HEAD [hunk 3]
```diff
@@ -0,0 +1,129 @@
+﻿using Akka.Actor;
```
Comment (rogeralsing) [inline src/contrib/dependencyInjection/Examples/BasicAutoFacUses/Program.cs~HEAD]: one more

File: src/contrib/dependencyInjection/Examples/BasicCastleWindsorUse/Actors.cs~HEAD [hunk 4]
```diff
@@ -0,0 +1,70 @@
+﻿﻿using Akka.Actor;
```
Comment (rogeralsing) [inline src/contrib/dependencyInjection/Examples/BasicCastleWindsorUse/Actors.cs~HEAD]: one more

File: src/contrib/dependencyInjection/Examples/BasicCastleWindsorUse/BasicCastleWindsorUses.csproj~HEAD [hunk 5]
```diff
@@ -0,0 +1,112 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (rogeralsing) [inline src/contrib/dependencyInjection/Examples/BasicCastleWindsorUse/BasicCastleWindsorUses.csproj~HEAD]: and more

**Comment by rogeralsing** [comment]
There are a lot of "*~Head" files in this PR. 
Are all conflicts resolved?
If so, those should be removed.

**Comment by Aaronontheweb** [comment]
Sigh.... build server isn't picking this one up. I'm going to have to mess with the timestamps again.

**Comment by Aaronontheweb** [comment]
@rogeralsing I resolved them - but might not have finished the rebase properly (might have skipped a patch - I had to do a lot of stuff manually afterwards.)

**Comment by jcwrequests** [comment]
@rogeralsing @arronontheweb sorry guys I was in the middle of trying to rebase and squash everything last night and had to stop before I was completely finished. Don't worry about losing anything because I have a local back up of my repo before I made all the changes. If you want me to delete my repo fork out again the make the changes the way I should have to begin with then I will certainly do that. @aarononthweb if you are able to fix it then I will leave it alone. I can even give you rights to the repo as well if that helps. I am real sorry for the confusion.

**Comment by jcwrequests** [comment]
Sorry I am doing this from my phone that should have been @aaronontheweb

**Comment by Aaronontheweb** [comment]
@jcwrequests you don't have anything to apologize for senor. I just wanted to get this cleaned up before I nuked the `dev` branch with all of the big RPs we're staging for V1. Your changes look good but I need input from some other contributors before I proceed, since these changes will be included as part of V1.

**Comment by jcwrequests** [comment]
@rogeralsing @aarononthrweb sorry guys I was in the middle of trying to rebase and squash everything last night and had to stop before I was completely finished. Don't worry about losing anything because I have a local back up of my repo before I made all the changes. If you want me to delete my repo fork out again the make the changes the way I should have to begin with then I will certainly do that. @aaronontheweb if you are able to fix it then I will leave it alone. I can even give you rights to the repo as well if that helps. I am real sorry for the confusion.

**Comment by stefansedich** [comment]
@Aaronontheweb as far as my original issue of memory leaks when using Autofac, I am happy that this PR fixes those issues for me.

**Comment by Aaronontheweb** [comment]
glad to hear it @stefansedich - please give this PR a good looking over and let me know if it's ok to merge.

**Comment by stefansedich** [comment]
Ok I am heading out for a bit but will give it a looking over in a few
hours.

Cheers

**Comment by jcwrequests** [comment]
@Aaronontheweb the head files should have been renamed and the regular .cs and .csproj should be been removed. I just fixed it on my end.Based on your description I am not sure if that is what you did.

**Comment by Aaronontheweb** [comment]
No worries @jcwrequests  - I cleaned it up.

**Comment by jcwrequests** [comment]
image
@Aaronontheweb Excellent!!!

File: src/contrib/dependencyInjection/Akka.DI.Core/DIActorContextAdapter.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
+﻿using Akka.Actor;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Akka.DI.Core
+{
+    public class DIActorContextAdapter
```
Comment (stefansedich) [inline src/contrib/dependencyInjection/Akka.DI.Core/DIActorContextAdapter.cs]: New lines are a bit sloppy on this file compared to the rest of the solution.

**Comment by stefansedich** [comment]
Looks good to me @Aaronontheweb I did notice though some sloppy new lines around the place, I left a comment on one file but I noticed it in many more (keep in mind I am fairly OCD when it comes to formatting so not sure if you guys care that much), apart from that it looks good to me.

**Comment by Aaronontheweb** [comment]
@stefansedich feel free to send a subsequent PR with formatting changes. That stuff is easy to review.

**Comment by Aaronontheweb** [comment]
Unless there are any objections, I'll rebase and merge this sucker tomorrow.

**Comment by stefansedich** [comment]
Ok sounds good to me!

**Comment by rogeralsing** [comment]
Status on this one? rebase and merge?

**Comment by Aaronontheweb** [comment]
@rogeralsing yep, I'll do that today - have one more PR of my own (build system stuff) I'm waiting on then I'll take care of this.

**Comment by Aaronontheweb** [comment]
Integrated the latest interface changes into @jcwrequests' DI fixes. Waiting on the build server to report in and then we'll merge.

**Pull request merged**